src/ParseObject.js

```diff
@@ -1363,7 +1363,7 @@ class ParseObject {
     try {
       await this.destroy(options);
     } catch (e) {
-      if (e.message === 'XMLHttpRequest failed: "Unable to connect to the Parse API"') {
+      if (e.message === CoreManager.get('CONNECTION_FAILED_MESSAGE')) {
```

Comment:
This seems like a super fragile design, testing against an error message. Unrelated to this PR, but that code really smells. Why is this not testing against the error code?

Reply:
The error message here can only be thrown from one place in the entire SDK. Error code 100 can have multiple error messages since it’s the fallback in the RESTController